docs/contributor/04-30-ci-cd.md

```diff
@@ -1,60 +1,46 @@
 # CI/CD
 The Istio module contains several pipelines that run on pull requests, on the main branch, on the release branch, and periodically. They perform various tasks such as running a linter, executing unit tests with code coverage, generating artifacts for development and production purposes, and running integration tests. The integration tests ensure the functional correctness of Kyma Istio Operator and verify if upgrading to the new version of Kyma Istio Operator is possible.
 
-## Pipelines Running on Pull Requests to the Main Branch
+## Enforcing that the required actions always run
+
+To ensure that the required actions always run, the Istio module uses GitHub ruleset.
+This enforces all jobs that check whether a particular workflow should start to always run.
+
+By default, Github Actions that don't run because of reasons other than skipping them in the `if` statement will have `Pending` status, and will not show up as a check in the Pull Request. The ruleset safeguard ensures that when it happens (for example because an external action is not allowed to run), the PR will not be able to be merged as it requires the check to run (either have status `Skipped` or `Success`)
```

Comment:
```suggestion
By default, GitHub Actions that don't run because of reasons other than being skipped in the `if` statement have the `Pending` status and do not appear as a check in a PR. The ruleset acts as a safeguard to prevent situations when, for example, an external action is not allowed to run. In such cases, the ruleset ensures that the PR cannot be merged unless the check is executed and has a status of either `Skipped` or `Success`.
```